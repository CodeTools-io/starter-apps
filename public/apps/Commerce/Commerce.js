import React, { lazy, Suspense } from 'react';
import { Route, Switch, useRouteMatch } from 'react-router-dom';

export default function Router() {
  const Cart = lazy(() => import('./Cart'));
  const Checkout = lazy(() => import('./Checkout'));

  const { path } = useRouteMatch();

  return (
    <Suspense fallback={<p>loading routeâ€¦</p>}>
      <Switch>
        <Route component={Cart} path={`${path}/cart`} />
        <Route component={Checkout} path={`${path}/checkout`} />
      </Switch>
    </Suspense>
  );
}
