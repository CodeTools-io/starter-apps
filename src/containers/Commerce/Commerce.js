import React from 'react';

import styles from './Commerce.module.css';

export default function Commerce ({ children, ...props }) {

  return <div className={styles.Commerce} {...props}>{children}</div>;
};



