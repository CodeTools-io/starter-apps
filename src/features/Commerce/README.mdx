---
type: module
id: Commerce
categoryId: feature
name: Commerce
path: /commerce
icon: Shop
---
