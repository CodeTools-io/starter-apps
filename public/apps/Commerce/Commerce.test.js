import React from 'react'
import Commerce from './Commerce'

describe('Commerce', () => {
  test('can render', () => {
    const { container } = renderContainer(
      <Commerce>
        <p>Content for the Commerce</p>
      </Commerce>
    )

    expect(container).toMatchSnapshot()
  })
})

