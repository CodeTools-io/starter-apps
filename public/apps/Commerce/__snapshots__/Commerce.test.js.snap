// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commerce can render 1`] = `
<div>
  <div
    class="ErrorBoundary"
  >
    <div
      class="StyledGrommet-sc-19lkkz7-0 lcXYAf"
    />
  </div>
</div>
`;
